DirectProgramming/DPC++FPGA/Tutorials/Features/loop_initiation_interval/src/loop_initiation_interval.cpp

```diff
@@ -0,0 +1,232 @@
+            num ^= 1;
+          }
+
+          int sum = 0;
+
+          // The intel::initiation_interval attribute is added here to "assert" that II=1 for
+          // this loop. Even though we fully expect the compiler to achieve
+          // II=1 here by default, some developers find it helful to include
+          // the attribute to "document" this expectation. If a future code
+          // change causes an unexpected II regression, the compiler will error
+          // out. Without the intel::initiation_interval attribute, an II regression may go
+          // unnoticed.
+#if defined(ENABLE_II)
+          [[intel::initiation_interval(1)]]
+#endif
+          // ---------------------------
+          // Long running loop
+          // ---------------------------
+          // The variable "sum" has a loop carried dependency with a feedback
+          // path, although not as long as the feedback path of "num". The first
+          // operation on "sum" in a given iteration depends on the value of
+          // "sum" calculated in the last operation of a previous iteration The
+          // compiler is able to achieve an II of 1 and the default targeted
+          // fMAX for Arria 10, but falls a little short on Stratix 10. The
+          // intel::initiation_interval attribute should not be used to relax the II of this loop
+          // as the drop in occupancy of the long loop is not worth achieving a
+          // slightly higher fMAX
+          for (size_t k = 0; k < kLongLoopSize; k++) {
+            sum += SomethingComplicated(num + k);
+          }
+
+          out_acc[i] = sum;
+        }
+      });
+    });
+
+    double exec_time = GetExecutionTime(e);
+    double inputMB = (kInputSize * sizeof(int)) / (double)(1024 * 1024);
+
+#if defined(ENABLE_II)
+    std::cout << "Kernel_ENABLE_II Throughput: " << (inputMB / exec_time)
+              << "MB/s\n";
+#else
+    std::cout << "Kernel Throughput: " << (inputMB / exec_time) << "MB/s\n";
+#endif
+    std::cout << "Exec Time: " << exec_time << "s , InputMB: " << inputMB
+              << "MB\n";
+
+  } catch (exception const &e) {
+    // Catches exceptions in the host code
+    std::cerr << "Caught a SYCL host exception:\n" << e.what() << "\n";
+
+    // Most likely the runtime couldn't find FPGA hardware!
+    if (e.get_cl_code() == CL_DEVICE_NOT_FOUND) {
+      std::cerr << "If you are targeting an FPGA, please ensure that your "
+                   "system has a correctly configured FPGA board.\n";
+      std::cerr << "Run sys_check in the oneAPI root directory to verify.\n";
+      std::cerr << "If you are targeting the FPGA emulator, compile with "
+                   "-DFPGA_EMULATOR.\n";
+    }
+    std::terminate();
+  }
+}
+
+int main() {
+  // seed random number generator
+  srand(0);
+
+  // input/output data
+  std::vector<int> in(kInputSize);
+  std::vector<int> out(kInputSize);
+
+  // Conservative max to avoid addition overflow
+  constexpr int randMax = 1 << 10;
```

Comment:
Inconsistant naming.   should either be kRandMax or rand_max

Reply:
This should be kRandMax actually, since it is a constexpr.